insights/parsers/softnet_stat.py

```diff
@@ -0,0 +1,100 @@
+number of times cpu woken up received_rps (tenth column), number of times reached flow
+limit count flow_limit_count (eleventh column).
+
+    The typical contents of ``cat /proc/net/softnet_stat`` file is as below::
+
+        00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+
+    Examples:
+        >>> SOFTNET_STATS = '''
+        ... 00008e78 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 000040ee 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0001608c 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ... 0000372f 00000001 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
+        ...'''.strip()
+        >>> softnet_obj = shared[SoftNetStats]
+        >>> softnet_obj.is_packet_drops()
+        True
+        >>> softnet_obj.get_cpu_instaces()
+        4
+        >>> softnet_obj.per_cpu_nstat('packet_drops')
+        {'cpu2': '00000000', 'cpu3': '00000001', 'cpu0': '00000000', 'cpu1': '00000000'}
+"""
+
+from .. import Parser, parser
+
+
+@parser("softnet_stat")
+class SoftNetStats(Parser):
+    """Parses ``/proc/net/softnet_stat`` file contains"""
+
+    def parse_content(self, contents):
+        cpu = 0
+        data = {}
+        for line in contents:
+            stats = line.split(' ')
+
+            data['cpu' + str(cpu)] = {'packet_process': stats[0],
+                                    'packet_drops': stats[1],
+                                    'time_squeeze': stats[2],
+                                    'cpu_collision': stats[8],
+                                    'received_rps': stats[9],
+                                    'flow_limit_count': stats[10]}
+            cpu = cpu + 1
+        self.data = data
+        self.data['cpu_instances'] = cpu
+
+    def get_cpu_instaces(self):
+        """
+        Get the number of cpu instances are handling network.
+
+        Arguments:
+            None: No input argument for the function.
+
+        Returns:
+            (int): The number of cpus handling network interrupts.
+        """
+        return self.data['cpu_instances']
+
+    def per_cpu_nstat(self, key):
+        """
+        Get network stats per column for all cpu.
+
+        Arguments:
+            (str): Column name for eg. packet_drops or received_rps.
+
+        Returns:
+            (dict): Column states per cpu.
+        """
+        data = {}
+        for each_cpu in self.data.keys()[1:]:
```

Comment:
Do you want to use the [1:]: to exclude the cpu_instances? The elements' order in dict is hashed, I do not suggest you do so.

Reply:
This would also be solved by not keeping the cpu_instances in the data property but making it a property of its own.